packages/nimble-components/src/select/index.ts

```diff
@@ -1307,6 +1316,27 @@ export class Select
         this.setPositioning();
         this.focusAndScrollOptionIntoView();
     }
+
+    private observeSelectedOptionTextContent(): void {
+        if (this.selectedOptionObserver) {
+            this.selectedOptionObserver.disconnect();
+        }
+
+        if (this.selectedIndex === -1) {
```

Comment:
Do you need this check? Is there a case where `this.firstSelectedOption` would return a truthy value when `this.selectedIndex === -1`?

Reply:
This is actually just handling a special case for when the `Select` initializes, whereby it sets `selectedIndex` to '-1'. At this point, the `selectedOptions` for the `Select` will be `undefined`, which without that check will cause an access of `firstSelectedOption` to throw.